apps/slack/slack_listeners.py

```diff
@@ -64,13 +65,17 @@ def _respond_to_message(event, channel_id, thread_ts, experiment_channel, experi
 
     if not session:
         external_id = make_session_external_id(channel_id, thread_ts)
-        session = SlackChannel.start_new_session(
-            working_experiment=experiment,
-            experiment_channel=experiment_channel,
-            participant_identifier=slack_user,
-            session_external_id=external_id,
-        )
-
+        session = ExperimentSession.objects.filter(external_id=external_id).first()
+        if not session:
+            try:
+                session = SlackChannel.start_new_session(
+                    working_experiment=experiment,
+                    experiment_channel=experiment_channel,
+                    participant_identifier=slack_user,
+                    session_external_id=external_id,
+                )
+            except IntegrityError:
+                session = ExperimentSession.objects.get(external_id=external_id)
```

Comment:
I don't think we should optimistically re-fetch the session. This seems like a rare edge case where 2 messages came in one after the other, the 2nd one before the 1st had completed.

What do you think of this approach:
```suggestion
          try:
              session = SlackChannel.start_new_session(
                  working_experiment=experiment,
                  experiment_channel=experiment_channel,
                  participant_identifier=slack_user,
                  session_external_id=external_id,
              )
          except IntegrityError as e:
              if external_id in str(e):
                  # make sure we only do this if there issue is the external ID
                  session = ExperimentSession.objects.get(external_id=external_id)
              else:
                  raise e
```